packages/@webex/plugin-cc/src/services/task/TaskManager.ts

```diff
@@ -0,0 +1,101 @@
+import EventEmitter from 'events';
+import {ICall, LINE_EVENTS} from '@webex/calling';
+import {WebSocketManager} from '../core/websocket/WebSocketManager';
+import routingContact from './contact';
+import WebCallingService from '../WebCallingService';
+import {ITask, TASK_EVENTS, TaskId} from './types';
+import {CC_EVENTS} from '../config/types';
+import {LoginOption} from '../../types';
+import Task from '.';
+
+export default class TaskManager extends EventEmitter {
+  private call: ICall;
+  private contact: ReturnType<typeof routingContact>;
+  private taskCollection: Record<TaskId, ITask>;
+  private webCallingService: WebCallingService;
+  private webSocketManager: WebSocketManager;
+  private static taskManager;
+  public task: ITask;
+
+  /**
+   * @param contact - Routing Contact layer. Talks to AQMReq layer to convert events to promises
+   * @param webCallingService - Webrtc Service Layer
+   * @param webSocketManager - Websocket Manager to maintain websocket connection and keepalives
+   */
+  constructor(
+    contact: ReturnType<typeof routingContact>,
+    webCallingService: WebCallingService,
+    webSocketManager: WebSocketManager
+  ) {
+    super();
+    this.contact = contact;
+    this.taskCollection = {};
+    this.webCallingService = webCallingService;
+    this.webSocketManager = webSocketManager;
+    this.registerTaskListeners();
+    this.registerIncomingCallEvent();
+  }
+
+  private registerIncomingCallEvent() {
+    this.webCallingService.on(LINE_EVENTS.INCOMING_CALL, (call) => {
+      if (this.task) {
+        this.emit(TASK_EVENTS.TASK_INCOMING, this.task);
+      }
+      this.call = call;
+    });
+  }
+
+  private registerTaskListeners() {
+    this.webSocketManager.on('message', (event) => {
+      const payload = JSON.parse(event);
+      if (!event.keepalive) {
+        switch (payload.data.type) {
+          case CC_EVENTS.AGENT_CONTACT_RESERVED:
+            this.task = new Task(this.contact, this.webCallingService, payload.data);
```

Comment:
One doubt, are we sure this will properly store the required task details from the payload? While testing I had found some issues, so I wanted to be a 100% sure.

Reply:
Could you please elaborate on what issue you came across?  Within the payload, data property is what we are passing which has complete event detail which will get saved as one of the property within the task object